compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -810,28 +873,161 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
+                            None,
+                            self.arena.alloc(hir::Path {
+                                res: Res::Def(
+                                    DefKind::TyParam,
+                                    def_id,
+                                ),
+                                span,
+                                segments: arena_vec![self; hir::PathSegment::from_ident(self.lower_ident(param.ident))],
+                            }),
+                        )),
+                    }),
+                    GenericParamKind::Const { .. } => {
+                        let parent_def_id = self.current_hir_id_owner;
+                        let node_id = self.next_node_id();
+
+                        self.create_def(
+                            parent_def_id,
+                            node_id,
+                            DefPathData::AnonConst,
+                        );
+                        let hir_id = self.lower_node_id(node_id);
+
+                        let span = self.lower_span(span);
+                        let ct = hir::AnonConst {
+                            hir_id,
+                            body: self.lower_body(|this| {
+                                (
+                                    &[],
+                                    hir::Expr {
+                                        hir_id: this.next_id(),
+                                        kind: hir::ExprKind::Path(hir::QPath::Resolved(
+                                            None,
+                                            this.arena.alloc(hir::Path {
+                                                res: Res::Def(
+                                                    DefKind::ConstParam,
+                                                    def_id,
+                                                ),
+                                                span,
+                                                segments: arena_vec![this; hir::PathSegment::from_ident(this.lower_ident(param.ident))],
+                                            }),
+                                        )),
+                                        span,
+                                    },
+                                )
+                            }),
+                        };
+
+                        hir::GenericArg::Const(hir::ConstArg { value: ct, span })
+                    }
+                }
+            })
+            .collect()
+    }
+
     /// Lowers a lifetime binder that defines `generic_params`, returning the corresponding HIR
     /// nodes. The returned list includes any "extra" lifetime parameters that were added by the
     /// name resolver owing to lifetime elision; this also populates the resolver's node-id->def-id
     /// map, so that later calls to `opt_node_id_to_def_id` that refer to these extra lifetime
     /// parameters will be successful.
```

Comment:
Could you complete the comment here?

Reply:
Done https://github.com/rust-lang/rust/pull/101345/commits/d45f1d3a83aedff4b4cca2611e8736d2dd916864